### ISSUE TITLE
ERROR 405: Method Not Allowed

### ISSUE THREAD

**Issue by rEnr3n** [AladW/aurutils#732]
__Before filing a bug, check the following:__

1) Do you use the latest release of aurutils and its dependencies? If so,
is the issue reproducible from the master branch?
```
$ pacman -Qs aurutils
local/aurutils 2.3.3-1
    helper tools for the arch user repository
```
I can't reproduce the issue on master branch.

2) Does the package conform to 
[`PKGBUILD(5)`](https://www.archlinux.org/pacman/PKGBUILD.5.html) and the 
[AUR package guidelines](https://wiki.archlinux.org/index.php/Arch_User_Repository#Submitting_packages)?
3) Does the package provide the correct metadata on the 
[AUR RPC interface](https://aur.archlinux.org/rpc.php)?

https://aur.archlinux.org/rpc/?v=5&type=search&by=name-desc&arg=yay

4) Does the package build with `makepkg -s` or `extra-x86_64-build` ?
5) Is the problem reproducible, and not due to a misconfiguration of
`pacman`, `makepkg`, `sudoers`, `gpg` or others?

I cannot replicate the issue on a clean container.

__If yes to all, create a debug log:__

```
$ AUR_DEBUG=1 aur search yay
https://aur.archlinux.org/rpc/?v=5&type=search&by=name-desc&arg=yay:
2020-08-10 16:30:22 ERROR 405: Method Not Allowed.
```

I'm not sure what the problem is as I have not changed any related configuration recently. I tried deleting folders in `~/.config` and `~/.cache` but I still have the same issue.

**Comment by rafasc** [comment]
I cannot reproduce the issue. Are you able to `curl` that url?
`$ curl 'https://aur.archlinux.org/rpc/?v=5&type=search&by=name-desc&arg=yay'`

**Comment by rEnr3n** [comment]
```
$ curl 'https://aur.archlinux.org/rpc/?v=5&type=search&by=name-desc&arg=yay'
{"version":5,"type":"search","resultcount":10,"results":[{"ID":544072,"Name":"puyo","PackageBaseID":135895,"PackageBase":"puyo","Version":"1.0-1","Description":"A frontend for pacman and yay.","URL":"https:\/\/github.com\/Appadeia\/puyo","NumVotes":5,"Popularity":0.002186,"OutOfDate":null,"Maintainer":null,"FirstSubmitted":1536534050,"LastModified":1536985328,"URLPath":"\/cgit\/aur.git\/snapshot\/puyo.tar.gz"},{"ID":698381,"Name":"pak-config-yay-git","PackageBaseID":149253,"PackageBase":"pak-config-yay-git","Version":"r3.f0ec020-1","Description":"A yay config for pak","URL":"https:\/\/gitlab.com\/moussaelianarsen\/pak","NumVotes":1,"Popularity":0.030168,"OutOfDate":null,"Maintainer":"Arsen6331","FirstSubmitted":1582007337,"LastModified":1582069308,"URLPath":"\/cgit\/aur.git\/snapshot\/pak-config-yay-git.tar.gz"},{"ID":698956,"Name":"aptyay-git","PackageBaseID":149233,"PackageBase":"aptyay-git","Version":"r2.42785d0-1","Description":"Changes yay syntax to be more like APT","URL":null,"NumVotes":1,"Popularity":0.028914,"OutOfDate":null,"Maintainer":"Arsen6331","FirstSubmitted":1581962068,"LastModified":1582188129,"URLPath":"\/cgit\/aur.git\/snapshot\/aptyay-git.tar.gz"},{"ID":731059,"Name":"ffpb","PackageBaseID":145346,"PackageBase":"ffpb","Version":"0.2.0-2","Description":"A progress bar for ffmpeg. Yay !","URL":"https:\/\/github.com\/althonos\/ffpb","NumVotes":2,"Popularity":0.585435,"OutOfDate":null,"Maintainer":"SleeplessSloth","FirstSubmitted":1569459159,"LastModified":1588632511,"URLPath":"\/cgit\/aur.git\/snapshot\/ffpb.tar.gz"},{"ID":767537,"Name":"yay-git","PackageBaseID":129573,"PackageBase":"yay-git","Version":"10.0.3.r0.ga8ba2a8-1","Description":"Yet another yogurt. Pacman wrapper and AUR helper written in go. (development version)","URL":"https:\/\/github.com\/Jguer\/yay","NumVotes":19,"Popularity":0.208941,"OutOfDate":null,"Maintainer":"jguer","FirstSubmitted":1517205142,"LastModified":1595230901,"URLPath":"\/cgit\/aur.git\/snapshot\/yay-git.tar.gz"},{"ID":767539,"Name":"yay","PackageBaseID":115973,"PackageBase":"yay","Version":"10.0.3-1","Description":"Yet another yogurt. Pacman wrapper and AUR helper written in go.","URL":"https:\/\/github.com\/Jguer\/yay","NumVotes":1192,"Popularity":63.175328,"OutOfDate":null,"Maintainer":"jguer","FirstSubmitted":1475688004,"LastModified":1595231161,"URLPath":"\/cgit\/aur.git\/snapshot\/yay.tar.gz"},{"ID":767540,"Name":"yay-bin","PackageBaseID":117489,"PackageBase":"yay-bin","Version":"10.0.3-1","Description":"Yet another yogurt. Pacman wrapper and AUR helper written in go. Pre-compiled.","URL":"https:\/\/github.com\/Jguer\/yay","NumVotes":131,"Popularity":6.696194,"OutOfDate":null,"Maintainer":"jguer","FirstSubmitted":1480777574,"LastModified":1595231194,"URLPath":"\/cgit\/aur.git\/snapshot\/yay-bin.tar.gz"},{"ID":773492,"Name":"pikaur-git","PackageBaseID":129630,"PackageBase":"pikaur-git","Version":"1.6.11-1","Description":"AUR helper which asks all questions before installing\/building. Inspired by pacaur, yaourt and yay.","URL":"https:\/\/github.com\/actionless\/pikaur","NumVotes":17,"Popularity":0.864605,"OutOfDate":null,"Maintainer":"actionless","FirstSubmitted":1517379650,"LastModified":1596439826,"URLPath":"\/cgit\/aur.git\/snapshot\/pikaur-git.tar.gz"},{"ID":773493,"Name":"pikaur","PackageBaseID":131064,"PackageBase":"pikaur","Version":"1.6.11-1","Description":"AUR helper which asks all questions before installing\/building. Inspired by pacaur, yaourt and yay.","URL":"https:\/\/github.com\/actionless\/pikaur","NumVotes":197,"Popularity":8.764067,"OutOfDate":null,"Maintainer":"actionless","FirstSubmitted":1521935919,"LastModified":1596439838,"URLPath":"\/cgit\/aur.git\/snapshot\/pikaur.tar.gz"},{"ID":776126,"Name":"yayim","PackageBaseID":154894,"PackageBase":"yayim","Version":"10.0.3_3-1","Description":"A modified version of yay with additional features, improvements and small bug fixes","URL":"https:\/\/git.jojii.de\/jojii\/yayim","NumVotes":0,"Popularity":0,"OutOfDate":null,"Maintainer":"Jojii","FirstSubmitted":1593342791,"LastModified":1596834585,"URLPath":"\/cgit\/aur.git\/snapshot\/yayim.tar.gz"}]}%
```

**Comment by rafasc** [comment]
Did you check if the problem persists, and this wasn't a momentary glitch from the RPC?

**Comment by rEnr3n** [comment]
Yes. I checked just now. Still the same.

**Comment by rafasc** [comment]
That version didn't use `curl`, it uses `wget`. Can you `wget` that url without getting the 405 error?
```
wget -O - -nv 'https://aur.archlinux.org/rpc/?v=5&type=search&by=name-desc&arg=yay'
```

Do you have a `.wgetrc`? I cannot reproduce this.
Could you perhaps use [xtrace](https://github.com/AladW/aurutils/blob/2.3/xtrace) and attach: `bash xtrace aur search yay`

because `AUR_DEBUG` mechanism didn't exist on `2.3.3`.

**Comment by rEnr3n** [comment]
Turns out the problem is this line on my `/etc/wgetrc`
```
content-disposition = on
```

I have commented out that line for now.

**Comment by AladW** [comment]
https://github.com/AladW/aurutils/commit/fd3ac21fd4c9c9730fd5708aba97ad42f027e018

**Comment by AladW** [comment]
Added some other commits to the 2.3 branch (in particular, removal of `parallel`), please test if it works as intended and I'll draft a release for 2.3.4.

**Comment by rEnr3n** [comment]
Tested the 2.3 branch. It works.

Using v2.3.3, I just wanted to add that using this config causes issue:
```
continue = on
content-disposition = on
```
Commenting either or all of the lines fixes the issue.

**Comment by rafasc** [comment]
Thanks for the follow up.

The manpage of `wget` itself marks `content-disposition` as experimental and refers that it may result in `HEAD` requests to the server, which the RPC server doesn't seem to support (upstream bug ?).

No action is needed on `master` because we moved to using `curl` in 84034bb (scripts: switch to curl, 2020-03-28) and `2.3` seems to work fine on my end. :rocket:

**Comment by AladW** [comment]
Released 2.3.4.
